app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java

```diff
@@ -59,99 +61,69 @@ public WifiSsidPreference(Context context) {
      */
     @Override
     protected void showDialog(Bundle state) {
-        WifiConfiguration[] networks = loadConfiguredNetworksSorted();
-        if (networks != null) {
-            Set<String> selected = getSharedPreferences().getStringSet(getKey(), new HashSet<>());
-            // from JavaDoc: Note that you must not modify the set instance returned by this call.
-            // therefore required to make a defensive copy of the elements
-            selected = new HashSet<>(selected);
-            CharSequence[] all = extractSsid(networks, false);
-            filterRemovedNetworks(selected, all);
-            setEntries(extractSsid(networks, true)); // display without surrounding quotes
-            setEntryValues(all); // the value of the entry is the SSID "as is"
-            setValues(selected); // the currently selected values (without meanwhile deleted networks)
-            super.showDialog(state);
-        } else {
-            Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_wifi_turn_on_wifi, Toast.LENGTH_LONG).show();
-        }
+        Context context = getContext();
+
+        Set<String> selected = getSharedPreferences().getStringSet(getKey(), new HashSet<>());
+        // from JavaDoc: Note that you must not modify the set instance returned by this call.
+        // therefore required to make a defensive copy of the elements
+        selected = new HashSet<>(selected);
+        List<String> all = new ArrayList<>(selected);
 
-        // On Android 8.1, ACCESS_COARSE_LOCATION is required, see issue #999
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
-            if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                if (getContext() instanceof Activity) {
-                    Activity activity = (Activity) getContext();
-                    ActivityCompat.requestPermissions(activity, new String[] { Manifest.permission.ACCESS_COARSE_LOCATION }, Constants.PERM_REQ_ACCESS_COARSE_LOCATION);
-                    this.getDialog().dismiss(); // wait for result
-                } else {
-                    Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_need_to_grant_location_permission, Toast.LENGTH_LONG).show();
+        boolean connected = false;
+        WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
+        if (wifiManager != null) {
+            WifiInfo info = wifiManager.getConnectionInfo();
+            if (info != null) {
+                String ssid = info.getSSID();
+                // api lvl 30 will have WifiManager.UNKNOWN_SSID
+                if (ssid != "" && !ssid.contains("unknown ssid") && !selected.contains(ssid)) {
```

Comment:
This lost the null check, and will blow up with NPE?

Reply:
I don't see what might be null here? We create `selected` ourselves - it may be empty but not null. And `ssid` is a string.

Reply:
Strings are nullable in java, so ssid.contains will blow up if ssid is null.

Reply:
The comment before even talks about ithttps://github.com/syncthing/syncthing-android/issues/620 in

Reply:
Audrius is right, I've had tests in the past I've documented in code where ssid was returned as null by Android. That happens when location is turned off.
See https://github.com/Catfriend1/syncthing-android/blob/d38f56f6967447680020eeaa3828367a4599d653/app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreferenceBase.java#L131

Reply:
Right, right, `String` is a class... Missed the comment because it was part of the code I just discarded due to `getConfiguredNetworks` deprecation. At least I know now why I asked for a review :|  
Thanks!